dali/pipeline/operators/crop/bbox_crop.h

```diff
@@ -0,0 +1,311 @@
+#define DALI_PIPELINE_OPERATORS_CROP_BBOX_CROP_H_
+
+#include <vector>
+#include <utility>
+#include <algorithm>
+
+#include "dali/common.h"
+#include "dali/error_handling.h"
+#include "dali/pipeline/operators/common.h"
+#include "dali/pipeline/operators/operator.h"
+
+namespace dali {
+
+class BBoxCrop : public Operator<CPUBackend> {
+  static const unsigned int kAttempts = 100;
+  static const unsigned int kBboxSize = 4;
+
+ protected:
+  struct Bounds {
+    explicit Bounds(std::vector<float> &&bounds)
+        : min(bounds[0]), max(bounds[1]) {
+      DALI_ENFORCE(min >= 0, "Min should be at least 0.0. Received: " +
+                                 std::to_string(min));
+      DALI_ENFORCE(bounds.size() == 2, "Bounds should be provided as 2 values");
+      DALI_ENFORCE(min <= max, "Bounds should be provided as: [min, max]");
+    }
+
+    bool Contains(float k) const { return k >= min && k <= max; }
+
+    const float min, max;
+  };
+
+  struct Rectangle {
+    explicit Rectangle(float left, float top, float right, float bottom)
+        : left(left),
+          top(top),
+          right(right),
+          bottom(bottom),
+          area((right- left) * (bottom - top)) {
+            // Enforce ltrb
+            DALI_ENFORCE(left >= 0 && left <= 1);
+            DALI_ENFORCE(top >= 0 && top <= 1);
+            DALI_ENFORCE(right >= 0 && right <= 1);
+            DALI_ENFORCE(bottom >= 0 && bottom <= 1);
+            DALI_ENFORCE(left <= right);
+            DALI_ENFORCE(top <= bottom);
+          }
+
+    bool Contains(float x, float y) const {
+      return x >= left && (x - left) <= right && y >= top &&
+             (y - top) <= bottom;
+    }
+
+    Rectangle ClampTo(const Rectangle &other) const {
+      const float new_left = std::max(other.left, left);
+      const float new_top = std::max(other.top, top);
+      const float new_right = std::min(other.right, right);
+      const float new_bottom = std::min(other.bottom, bottom);
+
+      return Rectangle(std::max(other.left, left), std::max(other.top, top),
+                       std::min(other.right, right),
+                       std::min(other.bottom, bottom));
+    }
+
+    Rectangle RemapTo(const Rectangle &other, unsigned int height, unsigned int width) const {
+      const float crop_width = other.right - other.left;
+      const float crop_height = other.bottom - other.top;
+
+      const float new_left = (std::max(other.left, left) - other.left) / crop_width;
+      const float new_top = (std::max(other.top, top) - other.top) / crop_height;
+      const float new_right = (std::min(other.right, right) - other.left) / crop_width;
+      const float new_bottom = (std::min(other.bottom, bottom) - other.top) / crop_height;
+
+      return Rectangle(
+          std::max(0.0f, std::min(new_left, 1.0f)), std::max(0.0f, std::min(new_top, 1.0f)),
+          std::max(0.0f, std::min(new_right, 1.0f)), std::max(0.0f, std::min(new_bottom, 1.0f)));
```

Comment:
You are not using `height` and `width` arguments.

Reply:
accepted